assets/src/js/menu_editor.js

```diff
@@ -0,0 +1,349 @@
+        markItemsOverflow(
+          errors,
+          children,
+          subLevelConf.maxItems,
+          sprintf(__('Sub menu "%s"', 'planet4-master-theme-backend'), title)
+        );
+      }
+
+      // Item title is too long
+      const titleCharsCount = [...title].length; // Count chars instead of code units
+      if (titleCharsCount > depthConf.maxChars) {
+        errors.push({
+          target: item.id,
+          message: sprintf(
+            __(
+              'This item label "%1$s" should not be longer than %2$s characters (currently %3$s).',
+              'planet4-master-theme-backend'
+            ),
+            title, depthConf.maxChars, titleCharsCount
+          )
+        });
+      }
+    });
+
+    // Display error messages and mark items in error
+    displayErrors(errors);
+  };
+
+  /**
+   * Get first checked location
+   *
+   * @return string The current location slug
+   */
+  const getCurrentLocation = () => {
+    return locations.map(location => {
+      return document.querySelector(`input[name="menu-locations[${location}]"]`).checked
+        ? location : null;
+    })[0] || null;
+  };
+
+  /**
+   * Gets the current conf.
+   *
+   * @return Object The current conf
+   */
+  const getCurrentConf = () => {
+    const location = getCurrentLocation();
+    const conf = p4_menu_config[location] || null;
+    if (!conf) {
+      return;
+    }
+
+    return {...{ maxItems: 5, maxChars: 18, maxDepth: 1, }, ...conf};
+  };
+
+  /**
+   * Get item children
+   *
+   * @param NodeElement item The item
+   * @return NodeElement[]|null Children list of this item
+   */
+  const getItemChildren = (item) => {
+    const depth = getItemDepth(item);
+    const itemsAfter = menuContent.querySelectorAll(
+      `#${item.id} ~ li.menu-item:not(.sortable-placeholder)`
+    );
+    if (itemsAfter.length <= 0 ) {
+      return null;
+    }
+
+    let children = [];
+    for (let child of [...itemsAfter]) {
+      if (!child.classList.contains(`menu-item-depth-${depth+1}`)) {
+        break;
+      }
+      children.push(child);
+    }
+
+    return children.length > 0 ? children : null;
+  };
+
+  /**
+   * Get the depth of the item
+   *
+   * @param NodeElement item The item
+   * @return int Depth of the item in the menu
+   */
+  const getItemDepth = (item) => {
+    const depthClass = [...item.classList]
+      .find(c => c.startsWith('menu-item-depth-'));
+    return parseInt(depthClass.substr(-1, 1));
+  };
+
+  /**
+   * Get the depth configuration (maxItems, maxChars)
+   * Defaults to global conf
+   *
+   * @param int depth The depth
+   * @return Object The depth conf
+   */
+  const getDepthConf = (conf, depth) => {
+    return (conf.depthConf && conf.depthConf[depth]) ? conf.depthConf[depth] : conf;
+  };
+
+  /**
+   * Change item depth, using nav-menu.js API
+   *
+   * @param NodeElement item The menu item
+   */
+  const normalizeDepth = (item, maxDepth) => {
+    let obj = jQuery(`#${item.id}`);
```

Comment:
`document.getElementById` is actually a simpler syntax :grin:

Reply:
The [nav-menu API](https://github.com/WordPress/WordPress/blob/master/wp-admin/js/nav-menu.js#L78) is not available if I only use getElementById, I don't know if there's another way to access it ?

Reply:
Ah didn't realize it needs jQuery, that's fine then :+1: